pkgs/development/compilers/llvm/common/default.nix

```
@@ -0,0 +1,742 @@
+          tools.clangUseLLVM
+        else if (pkgs.targetPackages.stdenv or args.stdenv).cc.isGNU then
+          tools.libstdcxxClang
+        else
+          tools.libcxxClang;
+
+      libstdcxxClang = wrapCCWith rec {
+        cc = tools.clang-unwrapped;
+        # libstdcxx is taken from gcc in an ad-hoc way in cc-wrapper.
+        libcxx = null;
+        extraPackages = [ targetLlvmLibraries.compiler-rt ];
+        extraBuildCommands = mkExtraBuildCommands cc;
+      };
+
+      libcxxClang = wrapCCWith rec {
+        cc = tools.clang-unwrapped;
+        libcxx = targetLlvmLibraries.libcxx;
+        extraPackages = [ targetLlvmLibraries.compiler-rt ];
+        extraBuildCommands = mkExtraBuildCommands cc;
+      };
+
+      lld = callPackage ./lld {
+        patches =
+          [ (metadata.getVersionFile "lld/gnu-install-dirs.patch") ]
+          ++ lib.optional (lib.versions.major metadata.release_version == "14") (
+            metadata.getVersionFile "lld/fix-root-src-dir.patch"
+          )
+          ++ lib.optional (
+            lib.versionAtLeast metadata.release_version "16" && lib.versionOlder metadata.release_version "18"
+          ) (metadata.getVersionFile "lld/add-table-base.patch");
+      };
+
+      lldb = callPackage ./lldb.nix (
+        {
+          patches =
+            let
+              resourceDirPatch = callPackage (
+                { substituteAll, libclang }:
+                (substituteAll {
+                  src = metadata.getVersionFile "lldb/resource-dir.patch";
+                  clangLibDir = "${libclang.lib}/lib";
+                }).overrideAttrs
+                  (_: _: { name = "resource-dir.patch"; })
+              ) { };
+            in
+            lib.optional (lib.versionOlder metadata.release_version "16")
+              # FIXME: do we need this after 15?
+              (metadata.getVersionFile "lldb/procfs.patch")
+            ++ lib.optional (lib.versionOlder metadata.release_version "17") resourceDirPatch
+            ++ lib.optional (lib.versionOlder metadata.release_version "14") (
+              metadata.getVersionFile "lldb/gnu-install-dirs.patch"
+            )
+            ++ lib.optional (lib.versionAtLeast metadata.release_version "14") ./lldb/gnu-install-dirs.patch
+            # This is a stopgap solution if/until the macOS SDK used for x86_64 is
+            # updated.
+            #
+            # The older 10.12 SDK used on x86_64 as of this writing has a `mach/machine.h`
+            # header that does not define `CPU_SUBTYPE_ARM64E` so we replace the one use
+            # of this preprocessor symbol in `lldb` with its expansion.
+            #
+            # See here for some context:
+            # https://github.com/NixOS/nixpkgs/pull/194634#issuecomment-1272129132
+            ++ lib.optional (
+              stdenv.targetPlatform.isDarwin
+              && !stdenv.targetPlatform.isAarch64
+              && (lib.versionOlder darwin.apple_sdk.sdk.version "11.0")
```

```suggestion
              && lib.versionOlder stdenv.targetPlatform.darwinSdkVersion "11.0"
```

will align this with the changes in https://github.com/NixOS/nixpkgs/pull/324155/files

then, which ever change has to deal with the merge conflicts with the deleted files can just re-delete the files and everything should just work.

---

Sweet, I've made the change. Not gonna push until #324155 is merged.